### PR TITLE
Blas migrator

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -303,6 +303,47 @@ def add_rebuild_libprotobuf(migrators, gx):
...
+        total_graph.predecessors(node)))
+    cycles = list(nx.simple_cycles(total_graph))
+    # print('cycles are here:', cycles)
+
+    migrators.append(
+        Rebuild(graph=total_graph,
+                pr_limit=5,
+                name='blas-2.0',
+                top_level=top_level,
+                cycles=cycles, obj_version=3))
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]:
```suggestion
                cycles=cycles, obj_version=0))
```
I think this is the first instance of this object.

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -303,6 +303,47 @@ def add_rebuild_libprotobuf(migrators, gx):
...
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(total_graph.selfloop_edges())
+
+    top_level = set(node for node in total_graph if not list(
+        total_graph.predecessors(node)))
+    cycles = list(nx.simple_cycles(total_graph))
+    # print('cycles are here:', cycles)
+
+    migrators.append(
+        Rebuild(graph=total_graph,
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]:
```suggestion
        BlasRebuild(graph=total_graph,
```
This has its own `Rebuild` subclass, right?

File: conda_forge_tick/migrators.xsh [hunk 3]
```diff
@@ -982,3 +982,79 @@ class ArchRebuild(Rebuild):
...
+            with open('meta.yaml', 'w') as f:
+                f.write(new_text)
+        return self.migrator_uid(attrs)
+
+    def pr_body(self):
+        body = super().pr_body()
+        additional_body = ("This PR has been triggered in an effort to update for new BLAS scheme.\n\n"
+                           "New BLAS scheme builds conda packages against the Reference-LAPACK's libraries\n"
+                           "which allows to change the BLAS implementation at install time by the user.\n\n"
+                           "which allows to change the BLAS implementation at install time by the user.\n\n"
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]:
```suggestion
```

File: conda_forge_tick/migrators.xsh [hunk 4]
```diff
@@ -982,3 +982,79 @@ class ArchRebuild(Rebuild):
...
+                           "the your rebuild has been merged.**\n\n"
+                           "This package has the following downstream children:\n"
+                           "{0}\n"
+                           "And potentially more."
+                           "".format('\n'.join([a[1] for a in list(self.graph.out_edges($PROJECT))[: 5]])))
+        body = body.format(additional_body)
+        return body
+
+    def commit_message(self):
+        return "bump build number"
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: Should this say something about blas?

File: conda_forge_tick/migrators.xsh [hunk 5]
```diff
@@ -982,3 +982,78 @@ class ArchRebuild(Rebuild):
...
+        body = body.format(additional_body)
+        return body
+
+    def commit_message(self):
+        return "Rebuild for new BLAS scheme"
+
+    def pr_title(self):
+        if self.name:
+            return 'Rebuild for new BLAS scheme'
+        else:
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: Is this clause needed?

File: conda_forge_tick/auto_tick.xsh [hunk 6]
```diff
@@ -303,6 +303,47 @@ def add_rebuild_libprotobuf(migrators, gx):
...
+        if not any([blas_c]):
+            pluck(total_graph, node)
+
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(total_graph.selfloop_edges())
+
+    top_level = set(node for node in total_graph if not list(
+        total_graph.predecessors(node)))
+    cycles = list(nx.simple_cycles(total_graph))
+    # print('cycles are here:', cycles)
```
Comment (scopatz) [inline conda_forge_tick/auto_tick.xsh]: Please remove dead code

**Comment by CJ-Wright** [comment]
Thank you very much!

**Pull request merged**